File: test-suite/elixir/guards_clean.ex
defmodule InventoryClean do
  @moduledoc """
  Every deep access sits behind an explicit nil guard.
  """

  def catalog_total(envelope) do
    case envelope do
      nil -> 0
      _ -> envelope.data.metrics.count.total
    end
  end

  def region_name(client) do
    if is_nil(client) do
      ""
    else
      client.conn.region.host.name
    end
  end

  def warehouse_slug(store) do
    unless is_nil(store) do
      store.primary.warehouse.code
    end
  end

  def region_name_with(client) do
    with {:ok, region} <- Map.fetch(client || %{}, :region) do
      region.host.name
    end
  end
end

File: test-suite/rust/guards_clean.rs
pub fn catalog_total(config: Option<&Config>) -> u64 {
    let Some(config) = config else {
        return 0;
    };
    config.runtime.limits.teams.max
}

pub fn catalog_total_if_let(config: Option<&Config>) -> u64 {
    if let Some(config) = config {
        return config.runtime.limits.teams.max;
    }
    0
}

pub fn shipment_region(client: Option<&Client>) -> String {
    match client {
        Some(client) => client.conn.region.host.name.clone(),
        None => String::new(),
    }
}

pub fn warehouse_slug(store: Option<&Store>) -> String {
    match store {
        Some(store) if store.primary.warehouse.code.is_empty() => String::new(),
        Some(store) => store.primary.warehouse.code.to_lowercase(),
        None => String::new(),
    }
}
